lib/content-services/src/lib/permission-manager/components/user-role-column/user-role-column.component.ts

```diff
@@ -80,10 +85,33 @@ export class UserRoleColumnComponent {
     placeholder: string = 'PERMISSION_MANAGER.LABELS.SELECT-ROLE';
 
     @Output()
-    roleChanged: EventEmitter<string> = new EventEmitter<string>();
+    roleChanged = new EventEmitter<string>();
+
+    i18nValue: string;
+
+    /* dropdown options, including i18n support */
+    options: RoleModelOption[] = [];
 
     onRoleChanged(newRole: string) {
         this.value = newRole;
         this.roleChanged.emit(newRole);
     }
+
+    private i18nRoleValue(value: string): string {
+        if (value) {
+            return `ADF.ROLES.${value.toUpperCase()}`;
+        }
+        return value;
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes.value) {
+            this.i18nValue = this.i18nRoleValue(changes.value.currentValue);
+        }
+
+        if (changes.roles) {
```

Comment:
Is the roles array something that will actively change while this component is 'alive' ? I'm asking because while the code will work without any issues, it does make it so that the roles input property is no longer directly accessed anywhere in this component, only indirectly accessed in ngOnChanges. 
If it won't change, then I guess shifting this code to ngOnInit/ngAfterViewInit or something similar should work too, and access the roles object directly
If it does change, then I guess we can leave it as proposed (although it does feel a bit weird to me to be honest)

Reply:
my intention was to change the source, where the roles are coming from, but this "roles" input is referenced with a bunch of other components that have it coming from the http responses. So no clue how often it's going to change. NgOnChanges is a good place as it is fired only when "roles" are changed